### PR TITLE
Graalvm json config with CI

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

- Based on https://github.com/awslabs/aws-crt-java/pull/749

*Description of changes:*

- Added CI for GraalVM on java 17, 21, 22. 
- Removed unused part from the original PR
- Fixed couple issue for GraalVM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/software/amazon/awssdk/crt/test/Http2StreamManagerTest.java [hunk 1]
```diff
@@ -167,7 +161,8 @@ public void testParallelRequestsWithLeakCheck(int numThreads, int numRequests) t
         // don't bother.
         // If we have a leak, we should have it on all platforms, and we'll catch it
         // elsewhere.
-        if (CRT.getOSIdentifier() != "android") {
+        // Disable the leak check for GraalVM as it's not using JVM, the assumption is not applicable.
+        if (CRT.getOSIdentifier() != "android" && System.getProperty("org.graalvm.nativeimage.imagecode") == null) {
```
Comment (DmitriyMusatkin) [inline src/test/java/software/amazon/awssdk/crt/test/Http2StreamManagerTest.java:165]: nit: disable comment is slightly confusing in this case as it is not enabled in the first place.

**Review by DmitriyMusatkin — APPROVED** [review]

File: .builder/actions/aws_crt_java_test.py [hunk 2]
```diff
@@ -40,18 +42,19 @@ def start_maven_tests(self, env):
 
         self._run_java_tests("-DrerunFailingTestsCount=5")
 
-        # run the ShutdownTest by itself
-        env.shell.setenv('AWS_CRT_SHUTDOWN_TESTING', '1')
-        self._run_java_tests("-Dtest=ShutdownTest")
+        if os.getenv("AWS_GRAAL_VM_CI") is None:
+            # run the ShutdownTest by itself
```
Comment (graebm) [inline .builder/actions/aws_crt_java_test.py:46]: document why we don't run this with GraalVM
also: why don't we run this with GraalVM?

File: .builder/actions/aws_crt_java_test.py [hunk 3]
```diff
@@ -11,6 +11,8 @@ def _run_java_tests(self, *extra_args):
             os.remove('log.txt')
 
         profiles = 'continuous-integration'
+        if os.getenv("AWS_GRAAL_VM_CI") is not None:
```
Comment (graebm) [inline .builder/actions/aws_crt_java_test.py:14]: utterly trivial/debatable:  GraalVM is always one word, feels weird as two
```suggestion
        if os.getenv("AWS_GRAALVM_CI") is not None:
```

File: src/test/java/software/amazon/awssdk/crt/test/Mqtt5ClientTestFixture.java [hunk 4]
```diff
@@ -18,9 +18,12 @@
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
+import org.junit.Assume;
+
 /* For environment variable setup, see SetupCrossCICrtEnvironment in the CRT builder */
 public class Mqtt5ClientTestFixture extends CrtTestFixture {
 
+    static final boolean AWS_GRAAL_VM_CI = System.getProperty("AWS_GRAAL_VM_CI") != null;
```
Comment (graebm) [inline src/test/java/software/amazon/awssdk/crt/test/Mqtt5ClientTestFixture.java:26]: trivial/same: 
```suggestion
    static final boolean AWS_GRAALVM_CI = System.getProperty("AWS_GRAALVM_CI") != null;
```

**Review by graebm — APPROVED** [review]

**Pull request merged**